ec2/spark_ec2.py

```diff
@@ -1259,6 +1259,15 @@ def real_main():
             cluster_instances=(master_nodes + slave_nodes),
             cluster_state='ssh-ready'
         )
+
+        # Determine types of running instances
+        existing_master_type = master_nodes[0].instance_type
+        existing_slave_type = slave_nodes[0].instance_type
+        if existing_master_type == existing_slave_type:
```

Comment:
Good point -- but the problem of `master_instance_type` differing from the slaves and handling the configs cleanly is outside the scope of this PR. 

Anyways I'm fine with setting it to `""` or not -- one thing you could do is add a comment saying why we do it as it seems unintuitive.